### DISCUSSION TITLE
hook event for item reorder

### DISCUSSION THREAD

**Discussion by valcriss** [directus/directus#10888] [Feature Requests]
Hello,

I think it will be interesting to have a hook event on item reorder.

Regards

**Comment by benhaynes** [comment]
CC: @nickrum @rijkvanzanten

**Reply by rijkvanzanten** [reply]
There's no special "item reorder" operation, so it'll be tricky to do a dedicated hook for it 🤔 

A "reorder" is nothing else than an update to set a "sort" field to a different value

**Reply by valcriss** [reply]
Hum, currently reordering an item do not trigger the items.update event hook. Is this a bug or is this possible in the future ?

**Reply by rijkvanzanten** [reply]
Ahh you mean the manual sorting on the layouts. I thought you meant in the relational interfaces. Nvm!

**Reply by infomiho** [reply]
Hey, just to surface this one more time. We have a need to purge external caching provider after items change. Since, the sort operation doesn't emit an event, it's not really possible to keep the cached response synced with the CMS.

It would be great if after this https://github.com/directus/directus/blob/0537663ab2d3df7b655e5dbe03794b9c83c56881/api/src/controllers/utils.ts#L79 there was an `items.update` or even `items.sorted` event that's triggered.

**Comment by infomiho** [comment]
@rijkvanzanten Here's an initial PR proposal for emitting an event after items are sorted. https://github.com/directus/directus/pull/11622
